server/src/main/java/org/elasticsearch/index/query/SpanMultiTermQueryBuilder.java

```diff
@@ -117,7 +147,44 @@ public static SpanMultiTermQueryBuilder fromXContent(XContentParser parser) thro
                     "[span_multi] must have [" + MATCH_FIELD.getPreferredName() + "] multi term query clause");
         }
 
-        return new SpanMultiTermQueryBuilder(subQuery).queryName(queryName).boost(boost);
+        return new SpanMultiTermQueryBuilder(subQuery).queryName(queryName).boost(boost).maxExpansions(maxExpansionValue);
+    }
+
+    public class TopTermSpanBooleanQueryRewriteWithMaxClause extends SpanMultiTermQueryWrapper.SpanRewriteMethod{
+        MultiTermQuery multiTermQuery;
```

Comment:
👍 , will do  -  I thought it will be nice to have  more dynamic way to override instead of boolean.max count ( I believe its a static property and requires yml file change ) ; Dynamically changing is very useful in case of cluster with multiple indices/data shapes. But I do agree that it can be confusing .

Reply:
> Dynamically changing is very useful in case of cluster with multiple indices/data shapes. 

Which `multi_term` query are you using ? Is it to increase the default value of 1024 ? Any multi term query that don't use top terms rewriting is not a good fit for the span query in general. If you need to perform prefix queries within span queries it is preferable to index the prefix using the `edge_ngram` filter. This way you can transform any prefix query into a span term query on a single term.

Reply:
Thanks for feedback @jimczi  , Multi queries could be prefix , wildcard or worst case  -regex. Im using span queries now for exact use case mentioned in ES documentation of span queries  - legal discovery. We considered edge ngrams which takes care of part of that use case. data volume( text)  and unpredictable cardinality  is another issue ( often few thousands for many multi searches) 

My request is to consider it as advanced param , but if elastic thinks it is confusing for users - i can change PR to use boolean query limit. Please let me know ! Thanks again

Reply:
My original intent was to add a `max_expansion` to force the rewrite of all multi terms to use a top terms rewrite when used inside a span query but in this case the `max_expansion` param can be interpreted as the number of top terms to keep during the rewrite. However I prefer your approach which throws an exception on `multi_term` queries that don't use top terms rewrite and matches more than 1024 terms. In this case we can fail the query with a nice message explaining that a top term rewrite should be used on the `multi_term` query if applicable. `max_expansion` is confusing in this scenario and doesn't bring much so I think we should simply honor the BooleanQuery max clause limit.

Reply:
👍  i will update the PR in a day or two

Reply:
thanks @nirmalc !